src/controllers/midi/midisourceclock.cpp

```diff
@@ -0,0 +1,117 @@
+#include "controllers/midi/midimessage.h"
+#include "controllers/midi/midisourceclock.h"
+#include "util/math.h"
+
+bool MidiSourceClock::handleMessage(unsigned char status) {
+    switch (status) {
+    case MIDI_START:
+        start();
+        return true;
+    case MIDI_STOP:
+        stop();
+        return true;
+    case MIDI_TIMING_CLK:
+        tick();
+        return true;
+    default:
+        return false;
+    }
+}
+
+void MidiSourceClock::start() {
+    m_bRunning = true;
+    m_iFilled = 0;
+    m_iRingBufferPos = 0;
```

Comment:
Yes! Here it is: 
http://www.blitter.com/~russtopia/MIDI/~jglatt/tech/midispec/seq.htm
> In other words, the MIDI Start puts the slave in "play mode", and the receipt of that first MIDI Clock marks the initial downbeat of the song (ie, MIDI Beat 0).